torch/_subclasses/fake_tensor.py

```diff
@@ -1232,6 +1271,14 @@ def dispatch(self, func, types, args=(), kwargs=None):
             # NOTE: incr is intentionally unused for a RAII pattern
             incr = IncrementRecursionCount()
 
+        # NOTE: Skip fakifying inputs to torch.ops.profiler because they only have c++ implementation
+        # and are performance critical. The suggested workaround will cause non-eligible overhead.
+        if func not in {
+            torch.ops.profiler._record_function_enter_new,
+            torch.ops.profiler._record_function_exit._RecordFunction,
+        }:
+            args, kwargs = _maybe_fakify_script_object(args, kwargs)
```

Comment:
This should not be where we're Fake-ifying the ScriptObject. For example, FakeTensorMode does not allow non-fake-Tensors by default (https://github.com/pytorch/pytorch/blob/c78f72d7e7965f8dc9516f462c2c78e140d4e57c/torch/_subclasses/fake_tensor.py#L824).

When we're doing make_fx tracing, make_fx should be fakeifying any ScriptObject inputs

Reply:
I tried that approach: i.e. fakifying the inputs like wrap. But this causes problem downstreams when cooperating with custom ops, because they expect the script object to be the real one in the schema.

Reply:
Do you know if the custom op expects a "TensorQueue" or if it expects a `torch::CustomClassHolder`?

Reply:
Using the FakeTensor-Tensor relationship as an analogy:
- FakeTensorQueue should be passable into C++.
- One should be able to call FakeTensorQueue methods in C++.
- Open question on what happens if TensorQueue has any public non-method members...

Reply:
I guess we need to ban public non-method members on these custom classes. Otherwise, there's really not going to be a way for us to write a "Fake" version of the class.

Reply:
> Do you know if the custom op expects a "TensorQueue" or if it expects a torch::CustomClassHolder?

It's expecting a TensorQueue.

> FakeTensorQueue probably needs to be a subclass (somehow) of TensorQueue ?

It seems that if we're going in this way, we are re-inventing tensor subclasses for torchbind classes? In the end probably requires extending the dispatcher infra as well?